task_processing/plugins/kubernetes/kubernetes_pod_executor.py

```diff
@@ -43,6 +44,7 @@
 from task_processing.plugins.kubernetes.utils import get_sanitised_kubernetes_name
 
 logger = logging.getLogger(__name__)
+k8s_client_configuration = Configuration.get_default_copy()
```

Comment:
we should probably UPPER_CASE this to make it more obvious that this is a global variable

Reply:
I also wonder if it's worth having this in a central location so we truly only ever make a single Configuration object (although, going down to 2 is still a massive improvement :p)

Reply:
I do like the idea of the central location, we could even use the same Configuration() object and even pass it on to the ApiClient()